### PR TITLE
MCO-998: sets up MCO repository for Snyk CI scans

### PR DESCRIPTION
**Pull request by cheesesashimi**
**- What I did**

In order to get a baseline setup for automated Snyk security scans on each PR, I added a .snyk file to our repository root to ignore certain directories and files.

**- How to verify it**

Once https://github.com/openshift/release/pull/47445 lands, one can run the security job that PR creates. The security job should pass.

**- Description for the changelog**
Enable automated Snyk security scans

### CODE REVIEW

File: .snyk [hunk 1]
```diff
@@ -0,0 +1,6 @@
+exclude:
+  global:
+    - "vendor/**"
+    - "**/*_test.go"
+    - "pkg/controller/render/hash.go"
```
Comment (sinnykumari) [inline .snyk:5]: For future reference and self-documentation, can we add one line comment on why we are excluding these go files.

Comment (cheesesashimi) [reply .snyk]: For posterity, Snyk provides a schema for adding things like reasons, expiration dates, etc. In practice, I couldn't get it to work correctly, so I just added comments to each line explaining the what and why.

**Comment by djoshy** [comment]
/lgtm
/approve

Thanks for figuring this out, Zack!

File: .snyk (hunk 1 above)
Comment (sinnykumari) [reply .snyk]: Looks great, thanks Zack!

**Pull request merged**